sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/_base.py

```diff
@@ -373,13 +411,61 @@ def _should_collect_stats(self):
             and not self._instrumentation_collection
         )
 
+
+    # check to see whether its the case of customer stats collection
+    def _should_collect_customer_statsbeat(self):
+        env_value = os.environ.get("APPLICATIONINSIGHTS_STATSBEAT_ENABLED_PREVIEW", "")
+        is_customer_statsbeat_enabled = env_value.lower() == "true"
+        # Don't collect customer statsbeat for instrumentation collection or customer statsbeat exporters
+        return (
+            is_customer_statsbeat_enabled
+            and not self._instrumentation_collection
+            and not self._is_customer_stats_exporter()
+        )
+
     # check to see if statsbeat is in "attempting to be initialized" state
     def _is_statsbeat_initializing_state(self):
         return self._is_stats_exporter() and not get_statsbeat_shutdown() and not get_statsbeat_initial_success()
 
     def _is_stats_exporter(self):
         return self.__class__.__name__ == "_StatsBeatExporter"
 
+    def _is_customer_stats_exporter(self):
+        return self.__class__.__name__ == "_CustomerStatsBeatExporter"
+
+    def _determine_client_retry_code(self, error) -> Tuple[RetryCodeType, Optional[str]]:
+        if hasattr(error, 'status_code') and error.status_code in [401, 403, 408, 429, 500, 502, 503, 504]:
+            return (error.status_code, _UNKNOWN)
+
+        if isinstance(error, ServiceRequestError):
+            error_message = str(error.message) if error.message else ""
+        else:
+            error_message = str(error)
+
+        error_message_lower = error_message.lower()
+        if 'timeout' in error_message_lower or 'timed out' in error_message_lower:
+            return (RetryCode.CLIENT_TIMEOUT, error_message)
+        return (RetryCode.CLIENT_EXCEPTION, error_message)
+
+    def _track_retry_items(self, envelopes: List[TelemetryItem], error) -> None:
+        if self._customer_statsbeat_metrics and self._should_collect_customer_statsbeat():
+            retry_code, message = self._determine_client_retry_code(error)
```

Comment:
Careful with `error` because sometimes it seems like it's an `Exception` and other times it's a custom error returned by Breeze. Please make sure to try to replicate those scenarios and see if this is being handled.